src/webserver.cpp

```diff
@@ -463,7 +463,7 @@ MHD_Result webserver::post_iterator(void *cls, enum MHD_ValueKind kind,
 
     try {
         if (filename == nullptr || mr->ws->file_upload_target != FILE_UPLOAD_DISK_ONLY) {
-            mr->dhr->set_arg(key, std::string(mr->dhr->get_arg(key)) + std::string(data, size));
+            mr->dhr->set_arg_flat(key, std::string(mr->dhr->get_arg(key)) + std::string(data, size));
         }
```

Comment:
I think we should split these two use cases. If filename is null, it means it is not a file we are processing, so we can manage the normal flow with multiple arguments. The only case where we want to keep appending is the case where you have a filename && file_upload_target is not FILE_UPLOAD_DISK_ONLY (so, basically, this argument is a file argument and we are trying to load the file into memory)

Reply:
Maybe a bit orthogonal to this change? I can file a separate issue.

Reply:
I think it is the same issue, because we kept the behaviors of URL style args parsing consistent across GET and POST. Without the above, they'd behave inconsistently.